pkg/controller/restore_controller.go

```diff
@@ -71,23 +71,25 @@ var nonRestorableResources = []string{
 }
 
 type restoreController struct {
-	namespace           string
-	restoreClient       arkv1client.RestoresGetter
-	backupClient        arkv1client.BackupsGetter
-	restorer            restore.Restorer
-	objectStoreConfig   api.CloudProviderConfig
-	bucket              string
-	pvProviderExists    bool
-	backupLister        listers.BackupLister
-	backupListerSynced  cache.InformerSynced
-	restoreLister       listers.RestoreLister
-	restoreListerSynced cache.InformerSynced
-	syncHandler         func(restoreName string) error
-	queue               workqueue.RateLimitingInterface
-	logger              logrus.FieldLogger
-	logLevel            logrus.Level
-	pluginRegistry      plugin.Registry
-	metrics             *metrics.ServerMetrics
+	namespace                   string
+	restoreClient               arkv1client.RestoresGetter
+	backupClient                arkv1client.BackupsGetter
+	backupStorageLocation       arkv1client.BackupStorageLocationsGetter
```

Comment:
I'm thinking now that I don't have to add this or the lister in the constructor.

Reply:
At this point I was thinking the backup here was being handled as one of the CRDs, and I needed to handle the location as a CRD too (since it is). I figured we would need to access its elements somewhere inside the restore function. But when I looked closer at the restore function it doesn't seem like that. Then I also realized the schedule isn not being handled this way.

Reply:
Yes. As I was answering your question I was realizing this is the reasonable approach. Thanks for confirming. I started already on that route and will have questions soon.